### PR TITLE
Add small test for http and test coverage metric

### PR DESCRIPTION
**Pull request by Adam0Brien**
## Describe your changes

- Added small test to internal/server/http
- Added test coverage metric to `make test`

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,9 +64,13 @@ clean:
 .PHONY: test
 # run all tests
 test:
```
Comment (josejulio) [inline Makefile:66]: Could you use this target in the github actions file?

**Review by josejulio — APPROVED** [review]
lgtm!

File: Makefile (hunk 1 above)
Comment (Adam0Brien) [reply Makefile:66]: I think so! I did a quick Stack overflow search and found [this](https://stackoverflow.com/questions/66918575/how-to-use-your-own-makefile-in-github-actions) Ill make the change now!

Comment (josejulio) [reply Makefile:66]: I think you would only need to change [this](https://github.com/project-kessel/inventory-api/blob/main/.github/workflows/build-test.yml#L27) line to `make test`

Comment (Adam0Brien) [reply Makefile:66]: I changed [this](https://github.com/project-kessel/inventory-api/blob/a9a0d0c6a1df3f869ca653230802b2c02c7e0ea9/.github/workflows/build-test.yml#L29) line and checked CI, I can see the total test coverage [here](https://github.com/project-kessel/inventory-api/actions/runs/10391528735/job/28774641459#step:6:61)

**Pull request merged**